iothub_client/src/iothub_client_retry_control.c

```diff
@@ -10,8 +10,9 @@
 #include "azure_c_shared_utility/optimize_size.h"
 #include "azure_c_shared_utility/xlogging.h"
 
-#define RESULT_OK           0
-#define INDEFINITE_TIME     ((time_t)-1)
+#define RESULT_OK                 0
+#define INDEFINITE_TIME           ((time_t)-1)
+#define DEFAULT_MAX_DELAY_IN_SECS 30
```

Comment:
C# has this set at 10s by default. See docs here for one example: https://github.com/Azure/azure-iot-sdk-csharp/blob/master/iothub/device/devdoc/retrypolicy.md
Just pointing this out, I think 30 or even higher is a good idea if the server or network is busy to save power, etc.

Reply:
So I think the csharp documentation is outdated then. The cshap code shows it's using 30 seconds for max delay. Shouldn't that be the case?
I also do agree 30 seconds seems more appropriate, although that is a empirical opinion.

Reply:
Follow up item for the minimum delay: https://github.com/Azure/azure-iot-sdk-c/issues/1470